### DISCUSSION TITLE
permanency && custom domains

### DISCUSSION THREAD

**Discussion by DoxrGitHub** [codesandbox/codesandbox-client#8101] [Q&A] [answered]
is there a way to set a custom domain? my bigger question:

**is this URL "permanent?"**

or does it change every two weeks or something? I want this to be a way to host stuff (I'm trying to host a post route or a rest API if that is the name, sorry kinda new to this stuff) and I don't want my API URL to change. I am OK with it shutting down as long as it turn back on upon a HTTP request (kinda like replit, but they're changing in a way that is unsuitable for me, so I'm trying to change what I regularly use to code, this seems like a good option if the URL is "permanent" and if I could set a custom domain (I don't NEED to have an official feature for that. I could just set up a cloaked web redirect record to a (sub)domain))

thanks!
doxr

**Comment by JamesACS** [comment, marked as answer]
Hello Doxr, 

CodeSandbox at present isn't set up as a permanent hosting service. Our Sandboxes run on virtual machines that will hibernate if not opened/interacted within a certain period of time. You may also find that the URL changes on rare occasions (for example, if we need to use an alternative domain for a period of time). 

We recommend using CodeSandbox to develop your project, then moving it to a hosting service once complete to avoid any interruption.

**Reply by DoxrGitHub** [reply]
thanks, that's good to know.
